### ISSUE TITLE
Proposal: Contribution to Refactoring Codebase to Gradle Convention Plugin

### ISSUE THREAD

**Issue by VahidGarousi** [AdamMc331/AndroidAppTemplate#78]
Hello! I'm interested in contributing to the project titled 'Refactoring Codebase to Gradle Convention Plugin: Enhancing Build Efficiency and Maintainability.' I've reviewed the project's goals, I believe I can make valuable contributions to this effort.

Before I start working on any changes, I wanted to discuss this with you to ensure my contributions align with the project's goals and roadmap. I'd appreciate any guidance or information you can provide to get started.

Looking forward to your feedback!

**Comment by VahidGarousi** [comment]
@AdamMc331 did you read this issue?

**Comment by AdamMc331** [comment]
Can you share some resources on this plugin you're referring to?

**Comment by VahidGarousi** [comment]
https://github.com/Kaaveh/ComposeNews/tree/kmp/build-logic



Mastering Dependency Management: Version Catalog & Convention Plugin at Scale 🚀 https://medium.com/proandroiddev/mastering-android-dependency-management-b94205595f6b

**Comment by AdamMc331** [comment]
@VahidGarousi Thanks for sharing! I'm open to changes in this template, but I want to understand a little more about what specifically we're changing. For example, this project already uses [Gradle Version Catalogs](https://github.com/AdamMc331/AndroidAppTemplate/blob/development/gradle/libs.versions.toml).

So I think the suggestion here is specifically the Gradle Convention Plugin, which is helpful if this project were to scale up to multiple modules as we could reduce boilerplate across different modules?

**Comment by VahidGarousi** [comment]
@AdamMc331 
Yes, you're correct! The Gradle Convention Plugin can be beneficial for this project, especially if you plan to scale up to multiple modules. It helps reduce boilerplate code across different modules by applying consistent conventions. It allows you to define common configurations and settings in one place, making it easier to manage and maintain this project as it grows. This approach enhances consistency and streamlines the development process across modules.

**Comment by AdamMc331** [comment]
That makes sense! I'm not super familiar with it, if you'd like to make a PR we can discuss it more then!
